discovery-provider/integration_tests/tasks/test_anchor_program_indexer.py

```diff
@@ -111,6 +122,62 @@ def test_parse_tx(app):
     assert owner_eth_address_hex == "0x25A3Acd4758Ab107ea0Bd739382B8130cD1F204d"
 
 
+@pytest.mark.asyncio
+async def test_fetch_metadata(app, mocker):
+    with app.app_context():
+        db = get_db()
+        redis = get_redis()
+
+    populate_mock_db(db, basic_entities, block_offset=3)
+
+    solana_client_manager_mock = create_autospec(SolanaClientManager)
+    cid_metadata_client_mock = create_autospec(CIDMetadataClient)
+
+    cid_metadata_client_mock.async_fetch_metadata_from_gateway_endpoints.return_value = (
+        mock_cid_metadata
+    )
+
+    anchor_program_indexer = AnchorProgramIndexer(
+        PROGRAM_ID,
+        ADMIN_STORAGE_PUBLIC_KEY,
+        LABEL,
+        redis,
+        db,
+        solana_client_manager_mock,
+        cid_metadata_client_mock,
+    )
+    parsed_tx = {
+        "tx_metadata": {
+            "instructions": [
+                {
+                    "instruction_name": "init_user",
+                    "data": Container(
+                        [("metadata", list(mock_cid_metadata.keys())[0])]
```

Comment:
can we extract `list(mock_cid_metadata.keys())[0]` into a named var so it's clearer what this is referring to?